### ISSUE TITLE
ERROR - Hosts: all subsequent calls will fallback to backened idp

### ISSUE THREAD

**Issue by everywan-dev** [aker-gateway/Aker#92]
Hello, I have aker as a FreeIPA to manage the users, but I also manage a part with Json.

The problem I have is with a cloudlinux centos machine where the ssh port is 2202:

2019-06-04 10:17:50,610 - INFO - Core: Starting up, user=my.user from=XXX.XXX.XXX.XXX:50364
2019-06-04 10:17:50,610 - INFO - IdPFactory: trying dynamic loading of module : Json
2019-06-04 10:17:50,611 - INFO - Json: loaded
2019-06-04 10:17:50,615 - ERROR - Hosts: all subsequent calls will fallback to backened idp, cache error: Error 111 connecting to localhost:6379. Connection refused.
2019-06-04 10:17:54,903 - INFO - Core: Starting session UUID f80fb09e-4801-448a-91de-7684c243f039 for user my.user to host XXX.XXX.XXX.XXX

**Comment by anazmy** [comment]
The error "Error 111 connecting to localhost:6379. Connection refused" is something else, this means redis is not started, have you installed, configured and started redis?

**Comment by everywan-dev** [comment]
yes, redis is intall on server, others machine is ok, the problem is only for this machine.
The port localhost:6379 what is it for? Do I have to open it in the client's firewall?

Regards

**Comment by everywan-dev** [comment]
The redis in aker server is active (running), do I have to give some kind of access to the centos firewall?

2019-06-07 12:35:08,377 - INFO - Core: Starting up, user=oriol.centelles from=MYPUBLICIP:46208
2019-06-07 12:35:08,377 - INFO - IdPFactory: trying dynamic loading of module : Json
2019-06-07 12:35:08,378 - INFO - Json: loaded
2019-06-07 12:35:08,383 - INFO - Hosts: loading hosts from cache
2019-06-07 12:35:15,794 - INFO - Core: Starting session UUID c81b61c5-5dee-49d8-af98-394d672817d0 for user oriol.centelles to host 217.150.212.11

Regards

**Comment by anazmy** [comment]
> The redis in aker server is active (running), do I have to give some kind of access to the centos firewall?
> 
> 2019-06-07 12:35:08,377 - INFO - Core: Starting up, user=oriol.centelles from=MYPUBLICIP:46208
> 2019-06-07 12:35:08,377 - INFO - IdPFactory: trying dynamic loading of module : Json
> 2019-06-07 12:35:08,378 - INFO - Json: loaded
> 2019-06-07 12:35:08,383 - INFO - Hosts: loading hosts from cache
> 2019-06-07 12:35:15,794 - INFO - Core: Starting session UUID c81b61c5-5dee-49d8-af98-394d672817d0 for user oriol.centelles to host 217.150.212.11
> 
> Regards

Let me clear the confusion first.

For the problem mentioned in your first post:
"The problem I have is with a cloudlinux centos machine where the ssh port is 2202"
This is a current limitation, where aker wont connect to a custom ssh port. It is tracked on issue #38 and to be addressed in a future version.

Secondly, for the other problem 
"Error 111 connecting to localhost:6379."
port 6379 is the port where redis server works on. Redis should be installed on the aker server itself and it should be listening on the mentioned port, is that the case?

You can check by running:
* On aker server, check redis service status
` systemctl status redis`

* On aker serer, heck if redis is listening on the mentioned port
`netstat -anlpt|grep 6379`

* Check if firewall/iptables is configured to allow connections.

Also, for the redis connection problem, it shouldn't stop aker from working it will use the files on disk instead of memory cache, so it will be slower if your environment is big.
